gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java

```diff
@@ -116,6 +136,22 @@ public void testAddSpec() throws Exception {
     Assert.assertFalse(this.specStore.exists(URI.create("dummy")));
   }
 
+  @Test
+  public void testUpdateSpec() throws Exception {
+    this.specStore.addSpec(this.flowSpec);
+    Assert.assertEquals(this.specStore.getSize(), 1);
+    Assert.assertTrue(this.specStore.exists(this.uri1));
+    //Updating
+
+    this.specStore.addSpec(this.flowSpecUpdated);
+    Assert.assertEquals(this.specStore.getSize(), 1);
+    Assert.assertTrue(this.specStore.exists(this.uri1));
+    FlowSpec updated = (FlowSpec)this.specStore.getSpecImpl(this.uri1);
+    Assert.assertEquals(updated.getConfigAsProperties().getProperty("properties.user.to.proxy"),"value1Updated");
+    Assert.assertEquals(updated.getConfigAsProperties().getProperty("properties.gobblin.flow.sourceIdentifier"),"value2Updated");
+    Assert.assertEquals(updated.getConfigAsProperties().getProperty("properties.gobblin.flow.destinationIdentifier"),"value3Updated");
+  }
```

Comment:
Adding a field which is unmodified and the value remains intact after update operation would be helpful.

Reply:
addressed